lib/taskana-data/src/main/java/pro/taskana/sampledata/SampleDataGenerator.java

```diff
@@ -146,20 +131,36 @@ private StringReader selectSchemaScript(String dbProductName, String schemaName)
      * @return a array with the corresponding scripts files
      */
     private String[] getScriptList() {
-        String[] script = {WORKBASKET, DISTRIBUTION_TARGETS, CLASSIFICATION, TASK, ATTACHMENT, WORKBASKET_ACCESS_LIST,
-            OBJECT_REFERENCE};
-        ArrayList<String> scriptsList = new ArrayList<>(Arrays.asList(script));
+        ArrayList<String> scriptsList = getDefaultScripts();
 
         try {
-            runner.runScript(new BufferedReader(
-                new InputStreamReader(this.getClass().getResourceAsStream(CHECK_HISTORY_EVENT_EXIST),
-                    StandardCharsets.UTF_8)));
-            runner.runScript(new BufferedReader(
-                new InputStreamReader(this.getClass().getResourceAsStream(CLEAR_HISTORY_EVENTS), StandardCharsets.UTF_8)));
+            runner.runScript(getScriptBufferedStream(CHECK_HISTORY_EVENT_EXIST));
+            runner.runScript(getScriptBufferedStream(CLEAR_HISTORY_EVENTS));
             scriptsList.add(HISTORY_EVENT);
         } catch (Exception e) {
             LOGGER.error("The HISTORY_EVENTS table is not created");
         }
         return scriptsList.toArray(new String[0]);
     }
+
+    static String parseAndReplace(LocalDateTime now, String script) {
+        return replaceRelativeTimeFunction(now,
+            getScriptBufferedStream(script).lines().collect(Collectors.joining(System.lineSeparator())));
+    }
+
+    static BufferedReader getScriptBufferedStream(String script) {
+        return Optional.ofNullable(SampleDataGenerator.class.getResourceAsStream(script)).map(
+            inputStream -> new BufferedReader(
+                new InputStreamReader(inputStream, StandardCharsets.UTF_8))).orElse(null);
+    }
+
+    static ArrayList<String> getDefaultScripts() {
```

Comment:
Please return List<String>

Reply:
if we going more into refactoring. let's return streams.

Reply:
i like that

Reply:
done